lib/tools/te_enum.h

```diff
@@ -144,6 +144,47 @@ extern int te_enum_translate(const te_enum_trn trn[],
                              int value, te_bool reverse,
                              int unknown_val);
 
+/**
+ * A mapping between a bit number and a value that corresponds to this bit.
+ *
+ * An array of mappings should end with TE_ENUM_BITMASK_CONV_END.
+ */
+typedef struct te_enum_bitmask_conv {
+    /** Bit number. */
+    unsigned int bit;
+    /** Value that is represented by the bit. */
+    uint64_t val;
+} te_enum_bitmask_conv;
+
+/** Terminating element of an enum bitmask conversion array. */
+#define TE_ENUM_BITMASK_CONV_END { .bit = UINT_MAX, .val = 0 }
+
+/**
+ * A variable-sized bitmask.
+ */
+typedef union te_enum_bitmask {
```

Comment:
What's the purpose of this union? Why can't you use just `uint64_t` ? Please note that ISO C explicitly forbids using unions for backdoor type conversion and it won't work on big-endian archs.